### DISCUSSION TITLE
Bagaimana cara mengisi missing value sesuai dengan aggregasi ?

### DISCUSSION THREAD

**Discussion by Baguspramudito** [teamalgoritma/community#800] [Q&A] [answered]
Pada saat mengerjakan capstone project F&B terdapat missing value data visitor pada jam sekitar 10-12. Saya ingin mengisi missing value tersebut berdasarkan nilai rata-rata dari setiap jam bukan rata-rata keseluruhan data. Contoh rata-rata visitor di jam 10 adalah 32 orang maka data tersebut yang akan menjadi nilai seluruh visitor yang missing di jam 10. Mohon bantuannya 

![image](https://github.com/teamalgoritma/community/assets/141464903/8349a367-6897-40d8-9c5a-c79223531b74)

**Comment by IrfanChairurrachman** [comment, marked as answer]
dear @Baguspramudito ,

Semisal kita mempunyai data seperti berikut
![image](https://github.com/teamalgoritma/community/assets/56690035/40d98b80-e2ea-4bbb-b3ab-f1cbc47340ff)

Untuk mengisi missing value berdasarkan rata-rata dari group tertentu kita dapat menggunakan `group_by()` untuk mengelompokkan grupnya, setelah itu kita tinggal `mutate()` dan `replace_na()` untuk mengisi nilai berdasarkan kelompok yang telah kita `group_by()`.

Berikut contoh implementasinya.
![image](https://github.com/teamalgoritma/community/assets/56690035/61b29644-fc00-4079-b225-4eb6d3560910)

Dengan konsep yang sama, ini dapat menyelesaikan permasalahan yang Anda hadapi.

Sekian jawaban dari kami. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.
